services/QuillLMS/app/models/user.rb

```diff
@@ -243,13 +243,7 @@ def subscription_authority_level(subscription_id)
   end
 
   def eligible_for_new_subscription?
-    if subscription
-      # if they have a subscription it must be a trial one
-      Subscription::TRIAL_TYPES.include?(subscription.account_type)
-    else
-      # otherwise they are good for purchase
-      true
-    end
+    !subscription || Subscription::TRIAL_TYPES.include?(subscription.account_type)
```

Comment:
I think(?) this is just a `nil` check, so consider `subscription.nil?`for readability?